### PR TITLE
bugfix: prevent memory leak caused by time.After

### PR DESCRIPTION
**Pull request by ianchen0119**
Reference: https://www.arangodb.com/2020/09/a-story-of-a-memory-leak-in-go-how-to-properly-use-time-after/

### CODE REVIEW

File: pkg/pfcpsim/pfcpsim.go [hunk 1]
```diff
@@ -180,11 +180,22 @@ func (c *PFCPClient) PeekNextHeartbeatResponse() (*message.HeartbeatResponse, er
...
+	var resMsg message.Message = nil
+	var err error = nil
+	delay := time.NewTimer(c.responseTimeout)
+	for {
+		select {
+		case msg := <-c.recvChan:
+			if !delay.Stop() {
+				<-delay.C
+			}
+			resMsg = msg
```
Comment (osinstom) [inline pkg/pfcpsim/pfcpsim.go:192]: How do you return a message recieved from the channel?

File: pkg/pfcpsim/pfcpsim.go [hunk 2]
```diff
@@ -180,11 +180,22 @@ func (c *PFCPClient) PeekNextHeartbeatResponse() (*message.HeartbeatResponse, er
...
 func (c *PFCPClient) PeekNextResponse() (message.Message, error) {
-	select {
-	case msg := <-c.recvChan:
-		return msg, nil
-	case <-time.After(c.responseTimeout):
-		return nil, NewTimeoutExpiredError()
+	var resMsg message.Message = nil
+	var err error = nil
+	delay := time.NewTimer(c.responseTimeout)
+	for {
```
Comment (osinstom) [inline pkg/pfcpsim/pfcpsim.go:186]: Do we need for loop here? It wasn't used before.

**Review by osinstom — CHANGES_REQUESTED** [review]

File: pkg/pfcpsim/pfcpsim.go (hunk 2 above)
Comment (ianchen0119) [reply pkg/pfcpsim/pfcpsim.go:186]: @osinstom 
In the case of the pfcp message received, it only change the value of resMsg but not return.
At mean while, it also checks the status of the timer, if the timer still running then send the event to channel for terminating the timer.
After that, handler will going to the select timer case, and check whether resMeg exists.

Comment (ianchen0119) [reply pkg/pfcpsim/pfcpsim.go:186]: Hi @osinstom 
Please help to review the reply when you feel free.
Thanks!

**Review by osinstom — APPROVED** [review]

**Review by ajaythakurintel — APPROVED** [review]

**Review by thakurajayL — APPROVED** [review]

**Comment by ianchen0119** [comment]
Hi @thakurajayL 
Please help to approve the CI execution when you feel free.
I solve some issues raised by the golangci-linter.

Thanks.

**Pull request merged**